### PR TITLE
Create username component

### LINKED ISSUE #568: Extract username to its own component
**Describe the solution you'd like**
Similar to the password component in `shared/form`, create a new component for a username that both the `Login` and `Register` components can use.

**Additional context**
Branch from `0.2.0`

### PR DESCRIPTION
**Pull request by RivaD2**
## Summary of change
Created the `Username` component with the following props:
```ts
type UsernameProps = {
  fieldName: string,
  class?: string,
  isInvalid: boolean,
  onChange: (event: any) => void,
  areCredentialsInvalid:boolean,
};
```
- I took the `Password` component and modeled the `Username` component after it. 
-  I removed `<TextField />` from `Login.tsx` and `Register.tsx` and dropped in `Username` component.
-  Used the `<TextField/>` as a guide for props and created event handler `onUsernameChanged()`

## Related issue

Closes #568

## Pull request checklist

Please keep this checklist in & ensure you have done the following:

- [x] Read, understood and adhered to our [contributing document](https://github.com/knjk04/book-project/blob/master/CONTRIBUTING.md).
  - [x] Ensure that you were first assigned to a relevant issue before creating this pull request
  - [x] Ensure code changes pass all tests

- [x] Read, understood and adhered to our [style guide](https://github.com/knjk04/book-project/blob/master/STYLEGUIDE.md). A lot of our code reviews are spent on ensuring compliance with our style guide, so it would save a lot of time if this was adhered to from the outset. 

- [x] Filled in the summary, context (if applicable) and related issue section. Replace the square brackets and its placeholder content with your contents. For an example, see any merged in pull request
  - [x] Included a screenshot(s) if a UI change was involved (it may look different on a reviewer's device)

- [x] Created a branch that has a descriptive name (what your branch is for in a few words and includes the issue number at the end, e.g. `test-reading-goal-123`

- [x] Set this pull request to 'draft' if you are still working on it

- [x] Resolved any merge conflicts

For any of the optional checkboxes (e.g. the screenshots one), still check it if it does not apply.

If in doubt, get in touch with us via our [Slack workspace](https://join.slack.com/t/teambookproject/shared_invite/zt-jcijyenp-JiKFGBv62FIPoFnvOW6Ubg)

### CODE REVIEW

**Comment by RivaD2** [comment]
@knjk04 Is it okay to merge the PR as is (once I edit it out of draft form)? Also, regarding testing, I have some experience but must admit I'm not as comfortable adding tests when there aren't some other tests to reference (just in case).

**Comment by knjk04** [comment]
@RivaD2 That's fine. Once you're ready for a review, feel free to request it. From a quick look, it seems fine, but I'll take a closer look and test it on my machine when you're ready

**Comment by RivaD2** [comment]
@knjk04 Marked it as ready for review. Thank you again for your help :)

File: frontend/src/shared/form/Username.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    <TextField
+      id="standard-basic" 
+      className={props.class}
+      label={fieldName} 
+      variant="outlined" 
+      required 
+      autoFocus 
+      error={isInvalid}
+      onChange={(event) => onChange(event.target.value)}
+      helperText={areCredentialsInvalid ? 'Please enter a username' : null}
```
Comment (knjk04) [inline frontend/src/shared/form/Username.tsx:39]: This is fine, but we could change this to the following instead?

```jsx
areCredentialsInvalid && 'Please enter a username'
```

**Review by knjk04 — CHANGES_REQUESTED** [review]
Thanks for working on this! This is great and works as expected. I've left one suggestion for making the code even more concise.

**Comment by knjk04** [comment]
I've pushed to your branch as well, so I'd recommend pulling first

File: frontend/src/shared/form/Username.tsx (hunk 1 above)
Comment (RivaD2) [reply frontend/src/shared/form/Username.tsx:39]: @knjk04 Thank you for your review and feedback :)  I would be happy to make the change. I'll make sure to do a pull first as well.

Comment (RivaD2) [reply frontend/src/shared/form/Username.tsx]: @knjk04 Change was made, we should be good to go:)

**Review by knjk04 — APPROVED** [review]
Thanks for making the change!

**Pull request merged**